dnf-behave-tests/common/lib/file.py

```diff
@@ -58,3 +72,28 @@ def copy_file(source, destination):
 
 def file_timestamp(filename):
     return os.path.getmtime(filename)
+
+def decompress_file_by_extension(src, dst_folder=None):
+    content = None
+    if src.endswith(".bz2"):
+        content = bz2.open(src, "rb").read()
+    elif src.endswith(".gz"):
+        content = gzip.open(src, "rb").read()
+    elif src.endswith(".xz"):
+        content = lzma.open(src, "rb").read()
+
+    if not content:
+        return False
+
+    if not dst_folder:
+        return content
+    else:
+        basename = os.path.basename(src)
+        dst = os.path.join(dst_folder, basename)
+        if src.endswith(".bz2"):
+            destination = dst[:-4]
+        else:
+            destination = dst[:-3]
+
+        open(destination, "wb").write(content)
+        return destination
```

Comment:
This function returns a mix of very different values depending on the situation (`False`, path to a file, file contents).

I think this one should return the decompressed file content if it was an archive or `None` if it is not. Then you can have another function that will call this one and store the content in the destination. That one could probably return the decompressed file path. Also, this second function is really just used in the new tests in `createrepo_c/steps/repodata.py`, so you can put it there.

Reply:
I like your suggestion much better than my version, will do, thanks.